android/app/src/main/kotlin/org/getlantern/lantern/model/LanternSessionManager.kt

```diff
@@ -50,7 +59,21 @@ class LanternSessionManager(application: Application) : SessionManager(applicati
     }
 
     override fun currency(): String {
-        return getCurrency()?.let { it.currencyCode } ?: "usd"
+        return getCurrency().currencyCode ?: "usd"
+    }
+
+    fun deviceCurrencyCode(): String {
+        val defaultLocale = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            Resources.getSystem().configuration.locales[0]
+        } else {
+            Resources.getSystem().configuration.locale
+        }
+        val deviceLocal = Currency.getInstance(defaultLocale).currencyCode.lowercase()
+        return if (supportedCurrencyCodes.contains(deviceLocal)) {
+            deviceLocal
+        } else {
+            "usd"
+        }
```

Comment:
If the currency associated with the primary locale isn't supported, but another on the Resources.getSystem().configuration.locales list is valid, should we try using it instead of defaulting to usd?

Reply:
Maybe we filter locales from Resources.getSystem().configuration.locales until we find one that is on the supported currency codes list?

Reply:
I think that is a good idea, Changing...

Reply:
Done

Reply:
Thanks for making that change!